src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/SymbolEquivalenceComparer.EquivalenceVisitor.cs

```diff
@@ -45,17 +45,18 @@ public bool AreEquivalent(ISymbol x, ISymbol y, Dictionary<INamedTypeSymbol, INa
                     throw new InvalidOperationException("Stack too deep.");
                 }
 #endif
+
                 // https://github.com/dotnet/roslyn/issues/39643 This is a temporary workaround sufficient to get existing tests passing.
                 // This component should be modified to properly deal with differences caused by nullability.
                 if (x is ITypeSymbol xType && y is ITypeSymbol yType && xType.IsDefinition != yType.IsDefinition)
                 {
                     if (x.IsDefinition)
                     {
-                        y = yType.WithNullableAnnotation(xType.NullableAnnotation);
```

Comment:
`.WithNullableAnnotation` and `.NullableAnnotation` are not yet available in CodeStyle layer. Follow-up work item tracks reverting these changes once they are available.

Reply:
The change in this file is exactly what I would expect to see (i.e. we should not need a follow-up item to correct it).